Segment-ComScore/Classes/SEGComScoreIntegration.m

```diff
@@ -34,20 +34,24 @@ - (instancetype)initWithSettings:(NSDictionary *)settings andComScore:(id)scorAn
         SCORPublisherConfiguration *config = [SCORPublisherConfiguration publisherConfigurationWithBuilderBlock:^(SCORPublisherConfigurationBuilder *builder) {
             // publisherId is also known as c2 value
             builder.publisherId = settings[@"c2"];
-            builder.publisherSecret = settings[@"publisherSecret"];
-            builder.applicationName = settings[@"appName:"];
-            builder.usagePropertiesAutoUpdateInterval = [settings[@"autoUpdateInterval"] integerValue];
-            builder.secureTransmission = [(NSNumber *)[self.settings objectForKey:@"useHTTPS"] boolValue];
-            builder.liveTransmissionMode = SCORLiveTransmissionModeLan;
-
-            if ([(NSNumber *)[self.settings objectForKey:@"autoUpdate"] boolValue] && [(NSNumber *)[self.settings objectForKey:@"foregroundOnly"] boolValue]) {
-                builder.usagePropertiesAutoUpdateMode = SCORUsagePropertiesAutoUpdateModeForegroundOnly;
-            } else if ([(NSNumber *)[self.settings objectForKey:@"autoUpdate"] boolValue]) {
-                builder.usagePropertiesAutoUpdateMode = SCORUsagePropertiesAutoUpdateModeForegroundAndBackground;
-            } else {
-                builder.usagePropertiesAutoUpdateMode = SCORUsagePropertiesAutoUpdateModeDisabled;
-            }
+            builder.secureTransmissionEnabled = [(NSNumber *)[self.settings objectForKey:@"useHTTPS"] boolValue];
```

Comment:
Looks like on[ page 14](https://app.box.com/s/csd2t9sb9z0m6gdadg852ysl4724jac8/file/545670443958), they actually specify that this should be `enableImplementationValidationMode`. That might be why you were getting errors. We should update this to make sure the downstream tool is processing this config correctly. 

`[[SCORAnalyticsSCORAnalyticsconfiguration]enableImplementationValidationMode];`

Reply:
Thanks for that. I actually didn't notice.
I tried enableImplementationValidationMode. It was very useful for debugging. 
Like when I added in code, I can see all the properties that were being added and why they were not added.
So, for now, I can see that when I call addLabels, it's indeed being sent to Comscore servers but calling addMetaData for contentAssets was not. 
But using `SCORStreamingAdvertisementMetadata` in video playbacks sent the data to comscore server.
It looks there is modification in code required  in properties handling and mapping especially with `SCORStreamingAdvertisementMetadata`. And also to answer your first question, we also need to revise the settings properties being sent by customers, to make sure we truly map everything. e.g Genre, Season, Program....

Reply:
Can you please elaborate on what modification in code is required as relates to `SCORStreamingAdvertisementMetadata`. Please see my Slack message about creating a shared doc with steps to reproduce, any debugging information, and next steps.